sklearn/model_selection/_split.py

```diff
@@ -1101,6 +1138,105 @@ def split(self, X, y=None, groups=None):
+        functionality of the split time series data samples is possible.
+        The `n_splits` value is used to calculate the number of splits
+        when creating train/test indices.
+        While normally the `n_splits` value would be provided by the user,
+        entering `walk_forward` as the value for the `n_splits` parameter in the
+        TimeSeriesSplit constructor would call this method in order to
+        determine an appropriate `n_splits` value for the "walk_forward" feature.
+        The "walk_forward" feature is defined by any train/test indices that
+        have the first element starting `0` and
+        progreses in increasing values throughout seperate train/test indices
+        until the ending value equals the `x_value` value minus 1
+
+        Read more in the :ref:`User Guide <time_series_split>`.
+        Read more about the "walk_forward" feature
+        in this GitHub Issue: https://github.com/scikit-learn/scikit-learn/issues/22523
+
+        .. versionadded:: 0.18
+
+        Parameters
+        ----------
+        x_value : int
+            First element of the np.arange array.
+            This is the ending value to which
+            the "walk forward" train/test indices go up to
+
+            .. versionchanged:: 0.22
+
+        max_train_size : int
+            Maximum size for a single training set.
+
+            .. versionadded:: 0.24
+
+        test_size : int
+            Used to limit the size of the test set. Defaults to
+            ``n_samples // (n_splits + 1)``, which is the maximum allowed value
+            with ``gap=0``.
+
+            .. versionadded:: 0.24
+
+        Examples
+        --------------
+        >>> x = np.arange(15)
+        >>> cv = TimeSeriesSplit(n_splits="walk_forward", x_shape=x.shape[0],
+            max_train_size=10 ,test_size=2)
+        >>> for train_index, test_index in cv.split(x):
+        ...     print("TRAIN: ", train_index, "TEST: ", test_index)
+        ...
+        TRAIN:  [0 1 2 3 4 5 6 7 8] TEST:  [ 9 10]
+        TRAIN:  [ 1  2  3  4  5  6  7  8  9 10] TEST:  [11 12]
+        TRAIN:  [ 3  4  5  6  7  8  9 10 11 12] TEST:  [13 14]
+
+        >>> x = np.arange(15)
+        >>> cv = TimeSeriesSplit(n_splits="walk_forward", x_shape=x.shape[0],
+            max_train_size=3, test_size=1)
+        >>> for train_index, test_index in cv.split(x):
+        ...     print("TRAIN: ", train_index, "TEST: ", test_index)
+        ...
+        TRAIN:  [0 1 2] TEST:  [3]
+        TRAIN:  [1 2 3] TEST:  [4]
+        TRAIN:  [2 3 4] TEST:  [5]
+        TRAIN:  [3 4 5] TEST:  [6]
+        TRAIN:  [4 5 6] TEST:  [7]
+        TRAIN:  [5 6 7] TEST:  [8]
+        TRAIN:  [6 7 8] TEST:  [9]
+        TRAIN:  [7 8 9] TEST:  [10]
+        TRAIN:  [ 8  9 10] TEST:  [11]
+        TRAIN:  [ 9 10 11] TEST:  [12]
+        TRAIN:  [10 11 12] TEST:  [13]
+        TRAIN:  [11 12 13] TEST:  [14]
+        """
+        x = np.arange(x_value)
```

Comment:
I think that we only need to have:

```python
def get_n_splits(self, X=None, y=None, groups=None):
    if isinstance(self.n_splits, str):
        return X.shape[0] - (self.max_train_size + self.test_size) + 1
    return self.n_splits
```

Reply:
Yep, this is a lot simpler and readable than the previous method